WordPress/openverse
# Create the moderation decision model

## Fixes



Fixes #3636 by @sarayourfriend 

## Description



This PR

- adds the moderation decision models (abstract and concrete for both media types) 
- adds a `decision` field on the report models
- deprecates the `status` field on the report models to use the decision model
  - usages of this field are replaced with the properties based on the `decision` field
  - this field can be removed in the future by the PR for #3642
- updates the admin UI for reports to account for changed fields
- refactors code to reduce duplication by moving common code to parent class
- fixes a validation issue in the `AbstractMediaReport` model (and update tests)

## Note

The model uses some different names to match the naming scheme used by other API models. These are small, positive changes and should not block merge.

## Testing Instructions


0. Check out the PR and run the dev server.
1. Observe the admin views for the reports and decisions.

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
